ingest-app/src/cmr/ingest/api/subscriptions.clj

```diff
@@ -152,6 +153,31 @@
         (get-unique-native-id context subscription))
       native-id)))
 
+(defn add-if-missing
+  [context subscription metadata user field value-fn]
+  (if-let [old-value (field metadata)]
+    subscription
+    (do
+      (println (value-fn context user))
+      (assoc subscription :metadata
+             (json/generate-string
+              (assoc metadata field (value-fn context user)))))))
+
+(defn add-subscriber-id-if-missing
+  "If subscriber EmailAddress is provided, use it. Else, get it from EDL."
+  [context subscription token-user]
+  (let [metadata (json/parse-string (:metadata subscription) true)]
+    (add-if-missing context subscription metadata token-user :SubscriberId
+      (fn [ctx user] user))))
```

Comment:
If this just returns the user, don't need to pass in an extra values to the lambda, `'(constantly user)` would be enough.

Reply:
👍